apps/definition/lib/definition/schema/type.ex

```diff
@@ -44,7 +44,29 @@ defmodule Definition.Schema.Type do
     end)
   end
 
-  @spec access_path() :: spec
+  @spec map() :: spec
+  def map() do
+    spec(fn
+      %{__struct__: _} -> false
+      x -> is_map(x)
+    end)
+  end
+
+  @spec list() :: spec
+  def list() do
+    spec(is_list())
+  end
+
+  @spec atom() :: spec
+  def atom() do
+    spec(is_atom())
+  end
+
+  @spec boolean() :: spec
+  def boolean() do
+    spec(is_boolean())
```

Comment:
i'll leave it to @bbalser or @jdenen since they've been deeper in Norm for longer: is it preferred to just directly call `spec(predicate_func())` directly when you're invoking Norm, or is it preferable to wrap these `spec/1` calls?

Reply:
Yeah, we were curious about this too. We were trying to follow precedent but may have understood that precedent incorrectly.